common/source/docs/common-esc-telemetry.rst

```diff
@@ -0,0 +1,98 @@
+
+Telemetry data may be conveyed to the autopilot either by a separate wire connection to an autopilot's UART RX pin or across the signal wire used to convey motor speed information (Bi-Directional DShot). Currently only BLHeli-32 ESCs provide this capability but it is possible for an ESC not to implement the BLHeli firmware and still provide this capability. ESC telemetry of motor rpm is especially useful for controlling the center frequency of harmonic notch filters to control noise. See :ref:`common-esc-telem-based-notch` for more information.
+
+.. note:: ArduPilot does not currently support the polling of the ESCs for telemetry data via throttle idle messages over the signal line in non DShot protocols.
+
+Connecting the ESCs Telemetry wire
+==================================
+
+.. image:: ../../../images/dshot-pixhawk.jpg
+    :target: ../_images/dshot-pixhawk.jpg
+    :width: 600px
+
+Connect all ESC telemetry wires to a single serial port's RX pin on the autopilot (above diagram uses Serial5 as an example).  A pin or wire for ESC telemetry is pre-soldered on most BLHeli32 ESCs. If the wire isn't pre-soldered you will need to solder it yourself. CubePilot serial port pinsouts can be found :ref:`here <common-thecube-overview>`.
+
+Set the following parameters to enable BLHeli32 telemetry feedback to the autopilot's serial port:
+
+- :ref:`SERIALx_PROTOCOL <SERIAL5_PROTOCOL>` 16 (= ESC telemetry) where "x" is the autopilot serial port number connected to the ESCs telemetry wire.  The mapping between serial port numbering and UART physical ports for you autopilot should be documented in its description page linked :ref:`here <common-autopilots>`.
+
+- :ref:`SERVO_BLH_TRATE <SERVO_BLH_TRATE>` defaults to 10 and normally does not need to be changed. this enables telemetry at a 10Hz update rate from the ESC.  If using the :ref:`harmonic notch feature <common-imu-notch-filtering>` this can be raised to 100.
+
+- :ref:`SERVO_BLH_POLES <SERVO_BLH_POLES>` defaults to 14 which applies to the majority of brushless motors and normally does not need to be changed.  Adjust as required if you're using motors with a pole count other than 14 to calculate true motor shaft RPM from ESC's e-field RPM.
+
+
+Bi-directional DShot
+====================
+
+[site wiki="rover"]
+.. warning::
+
+   Bi-directional DShot does not work with Rover (`see issue here <https://github.com/ArduPilot/ardupilot/issues/20372>`__)
```

Comment:
I know this is existing text that has just been moved but in any case, this Bi-directional DShot issue for Rover has been fixed now.  https://github.com/ArduPilot/ardupilot/issues/20372

Reply:
will do